scripts/main/photo.js

```diff
@@ -564,173 +559,151 @@ photo.setStar = function (photoIDs, isStarred) {
+		formElements.is_public.nextElementSibling.textContent = lychee.locale["PHOTO_PUBLIC_EXPL"];
+		formElements.grants_full_photo.previousElementSibling.textContent = lychee.locale["PHOTO_FULL"];
+		formElements.grants_full_photo.nextElementSibling.textContent = lychee.locale["PHOTO_FULL_EXPL"];
+		formElements.requires_link.previousElementSibling.textContent = lychee.locale["PHOTO_HIDDEN"];
+		formElements.requires_link.nextElementSibling.textContent = lychee.locale["PHOTO_HIDDEN_EXPL"];
+		formElements.is_downloadable.previousElementSibling.textContent = lychee.locale["PHOTO_DOWNLOADABLE"];
+		formElements.is_downloadable.nextElementSibling.textContent = lychee.locale["PHOTO_DOWNLOADABLE_EXPL"];
+		formElements.is_share_button_visible.previousElementSibling.textContent = lychee.locale["PHOTO_SHARE_BUTTON_VISIBLE"];
+		formElements.is_share_button_visible.nextElementSibling.textContent = lychee.locale["PHOTO_SHARE_BUTTON_VISIBLE_EXPL"];
+		formElements.has_password.previousElementSibling.textContent = lychee.locale["PHOTO_PASSWORD_PROT"];
+		formElements.has_password.nextElementSibling.textContent = lychee.locale["PHOTO_PASSWORD_PROT_EXPL"];
+
+		if (photo.json.is_public === 2) {
+			// Public album.
+			dialog.querySelector("p#ppp_dialog_no_edit_expl").textContent = lychee.locale["PHOTO_NO_EDIT_SHARING_TEXT"];
+			dialog.querySelector("p#ppp_dialog_global_expl").remove();
+			// Initialize values of detailed settings according to album
+			// settings and hide action button as we can't actually change
+			// anything.
+			formElements.is_public.checked = true;
+			formElements.is_public.disabled = true;
+			formElements.is_public.parentElement.classList.add("disabled");
+			if (album.json) {
+				formElements.grants_full_photo.checked = album.json.grants_full_photo;
+				// Photos in public albums are never hidden as such.  It's the
+				// album that's hidden.  Or is that distinction irrelevant to end
+				// users?
+				formElements.requires_link.checked = false;
+				formElements.is_downloadable.checked = album.json.is_downloadable;
+				formElements.is_share_button_visible = album.json.is_share_button_visible;
+				formElements.has_password.checked = album.json.has_password;
 			}
-		});
-
-		if (photo.json.is_public === 1) {
-			$('.basicModal .switch input[name="is_public"]').click();
+			basicModal.hideActionButton();
+		} else {
+			// Private album
+			dialog.querySelector("p#ppp_dialog_no_edit_expl").remove();
+			dialog.querySelector("p#ppp_dialog_global_expl").textContent = lychee.locale["PHOTO_EDIT_GLOBAL_SHARING_TEXT"];
+			// Initialize values of detailed settings according to global
+			// configuration.
+			formElements.is_public.checked = photo.json.is_public !== 0;
+			formElements.grants_full_photo.checked = lychee.full_photo;
+			formElements.requires_link.checked = lychee.public_photos_hidden;
+			formElements.is_downloadable.checked = !!album.downloadable;
```

Comment:
Yes and no.

Bo, `lychee.downloadable` is also wrong, because it online affects public photos in an private album.

Yes, `album.downloadable` is only meaningful for public albums. The idea here is that the dialog shows the settings which would apply, if the album was public. The fact that `album.downloadable` has no effect is visualized by the control elements being greyed out.

Reply:
> Bo, lychee.downloadable is also wrong, because it online affects public photos in an private album.

I'm not sure if I understood you correctly given the typos, but this _is_ a case of a private album, hence we should be consulting `lychee.downloadable`.

> The idea here is that the dialog shows the settings which would apply, if the album was public.

No. For private albums, the dialog is supposed to show the global settings (`lychee.*`), since they are the ones applicable to public photos in private albums. album-level settings are irrelevant in that case and would only add to confusion.

I think I'm the one who coded this dialog in the first place, so let me summarize how I intended it to behave:
* If the album is public, we mostly mirror the album settings here, but everything is greyed out/disabled. There is a message at the top that explains why none of the settings can be changed.
* If the album is private but the photo is public, we mostly mirror the global settings here, but everything is again greyed out (apart from the "public" switch at the top) and there's a message in the middle that explains why.
* If the album is private *and* the photo is private, the (still greyed out) checkboxes are empty, since no settings apply in that case (the photo is private). From reading the code, your code does not handle that case though and it will fill in the checkboxes with global settings. I'm not sure if I like that but I reserve the judgement until I've had a chance to actually try it out.

Reply:
On thinking about it a little more, that last case above (of private album _and_ private photo) perhaps _should_ show global settings (as it does in your case), since those settings _will_ be applicable if the album is shared with another user...

Reply:
> > No, `lychee.downloadable` is also wrong, because it online affects public photos in an private album.
> 
> I'm not sure if I understood you correctly given the typos, but this _is_ a case of a private album, hence we should be consulting `lychee.downloadable`.

Yes, it is a private _album_. The point is, the album dialog shows the _album_ settings, not the settings which apply to a public photo in that private album. If users want to see the settings of a photo, the need to open the photo and look at the photo settings of that.

> > The idea here is that the dialog shows the settings which would apply, if the album was public.
> 
> No. For private albums, the dialog is supposed to show the global settings (`lychee.*`), since they are the ones applicable to public photos in private albums. album-level settings are irrelevant in that case and would only add to confusion.

I guess this is the point were we are on different pages. I believe the album dialog should always show the settings of the album, even if some of the settings are currently irrelevant which is visually indicated by greyed out controls. Personally, I would deem it confusing, if the album dialog happens to show (global) photo settings only because some album settings are not applicable at the moment.
 
> I think I'm the one who coded this dialog in the first place, so let me summarize how I intended it to behave:
> 
>     * If the album is public, we mostly mirror the album settings here, but everything is greyed out/disabled. There is a message at the top that explains why none of the settings can be changed.

OK, and here is the point where I notice that you are speaking about the "photo dialog" and not the "album dialog" as I could have noticed by looking at the file name `photos.js`. I am really worn out.